### DISCUSSION TITLE
[Native DLL] Fixed C strings

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3020] [General]
How to handle fixed size C strings in structs?

**Comment by SunSerega** [comment]
The easiest way is `[MarshalAs]` on fields.
But it will copy (and convert) the string data on every external call. And you need to be careful with the right combination of `[In]` and `[Out]` attributes on the subprogram that takes in a record (struct) with such a field, to set the right marshaling direction.

For my own native wrappers, I code-gen things like this:
https://github.com/SunSerega/POCGL/blob/e1b884ae85c97613c8e095c0f714b63666d9f2e1/Packing/Template/LowLvl/OpenCL/Types.Interface.template#L5834
